main/src/main/scala/sbt/internal/server/LanguageServerProtocol.scala

```diff
@@ -60,6 +58,12 @@ private[sbt] trait LanguageServerProtocol extends CommandChannel {
         langRespond(InitializeResult(serverCapabilities), Option(request.id))
       case "textDocument/didSave" =>
         append(Exec("compile", Some(request.id), Some(CommandSource(name))))
+      case "textDocument/definition" =>
+        import sjsonnew.support.scalajson.unsafe.CompactPrinter
+        append(
+          Exec(s"lspDefinition ${CompactPrinter(json)}",
```

Comment:
The query for definition, if possible, should happen concurrently with whatever that's happening in the command processing engine. For example, the user could be in the middle of `test` that takes 10 min. I think `textDocument/definition` should return while the test is running, not afterwards. So I don't think we should use task or input task here.

Reply:
looks like fixed I hope